### PR TITLE
MODINVSTOR-376: Handle statusUpdatedDate property for instance.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/MODINVSTOR-376: Instance record. Status updated date does not populate in detailed view

### Approach:

Added new before update trigger for instance table that sets `statusUpdatedDate` = NOW in case instance's `statusId` property has been changed.

The date time in `statusUpdatedDate` has the following format `yyyy-MM-dd'T'HH:mm:ss.SSS` (ISO-8601 that is standard in java) and it is in UTC timezone.

### CODE REVIEW

File: src/test/java/org/folio/rest/api/InstanceStorageTest.java [hunk 1]
```diff
@@ -1518,6 +1534,97 @@ public void shouldReturnErrorResponseIfAllInstancesFailed()
...
+
+    assertThat(updatedInstance
+        .getString(STATUS_UPDATED_DATE_PROPERTY), withinSecondsBeforeNow(seconds(2)));
+  }
+
+  /**
+   * Test case for instanceStatusUpdatedDateTrigger.sql trigger.
+   */
+  @Test
+  public void shouldOverrideStatusUpdatedDate() throws Exception {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/InstanceStorageTest.java]: What does `override` mean in this context, is this test about the status updated date changing when the status changes?

Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/api/InstanceStorageTest.java]: this test about updating instance status twice.

Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/api/InstanceStorageTest.java]: @bohdan-suprun It is about the `status updated date` being updated when the status changes for a second time?

If so, maybe something like `shouldChangeStatusUpdatedDateOnSubsequentStatusChanges`?

File: src/test/java/org/folio/rest/api/InstanceStorageTest.java [hunk 2]
```diff
@@ -1518,6 +1534,97 @@ public void shouldReturnErrorResponseIfAllInstancesFailed()
...
+        .getString(STATUS_UPDATED_DATE_PROPERTY), withinSecondsBeforeNow(seconds(2)));
+    assertThat(updatedInstanceWithOthStatus
+        .getString(STATUS_UPDATED_DATE_PROPERTY), withinSecondsBeforeNow(seconds(1)));
+  }
+
+  /**
+   * Test case for instanceStatusUpdatedDateTrigger.sql trigger.
+   */
+  @Test
+  public void shouldNotOverrideStatusUpdatedDateOnSameStatusId() throws Exception {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/InstanceStorageTest.java]:
```suggestion
  public void shouldNotChangeStatusUpdatedDateWhenStatusHasNotChanged() throws Exception {
```

**Pull request merged**